instrumentation/spring/spring-cloud-aws-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/aws/MessagingMessageListenerAdapterInstrumentation.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.aws;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+
+import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeTransformer;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.springframework.messaging.Message;
+
+public class MessagingMessageListenerAdapterInstrumentation implements TypeInstrumentation {
+
+  @Override
+  public ElementMatcher<TypeDescription> typeMatcher() {
+    return named("io.awspring.cloud.sqs.listener.adapter.MessagingMessageListenerAdapter");
+  }
+
+  @Override
+  public void transform(TypeTransformer transformer) {
+    transformer.applyAdviceToMethod(
+        named("onMessage").and(takesArgument(0, named("org.springframework.messaging.Message"))),
```

Comment:
https://github.com/awspring/spring-cloud-aws/blob/0882fb17899a8bd448014603f366afdefe6e6fc7/spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/listener/adapter/MessagingMessageListenerAdapter.java#L45
for it, don't we need to support? If yes, is it necessary to comment here?

Reply:
added a `TODO`. I skipped it because I wasn't quite sure how to handle it.